src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -835,7 +873,21 @@ public void DefaultExecuteCmdlet()
                 this.VM.Identity = new VirtualMachineIdentity(null, null, Microsoft.Azure.Management.Compute.Models.ResourceIdentityType.SystemAssigned);
             }
 
+            if (this.VM.SecurityProfile.SecurityType == "TrustedLaunch" || this.VM.SecurityProfile.SecurityType == "ConfidentialVM")
```

Comment:
there shuld be null checks here, likely the ? at the end of each property in 'this' except SecurityType.

Reply:
Added ? at the end. No null checks needed.